### PR TITLE
refactor(utility): set "LazyFactory" as data manager

### PR DESCRIPTION
**Pull request by linjiX**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2846jb2 将 LazyFacrory 设置为数据管理中心](https://app.clickup.com/t/2846jb2)

File: graviti/utility/lazy2.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+    def __init__(
+        self,
+        factory: "LazyFactory",
+        keys: Tuple[str, ...],
+    ) -> None:
+        self._total_count = factory._total_count
+        self._keys = keys
+        self._limit = factory._limit
+        self._pages: List[Union["LazyPage", pa.Array]] = [
```
Comment (Lee-000) [inline graviti/utility/lazy2.py:32]: Why the double quotes around `LazyPage`?

**Review by AChenQ — APPROVED** [review]

File: graviti/utility/lazy2.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        ]
+        self._factory = factory
+
+    def __len__(self) -> int:
+        return self._total_count
+
+    def __iter__(self) -> Iterator[Any]:
+        return chain(*self._pages)
+
+    def __getitem__(self, index: int) -> Any:
```
Comment (Lee-000) [inline graviti/utility/lazy2.py:43]: Is the return type `Any` or `pyarrow.Scalar`?

Comment (linjiX) [reply graviti/utility/lazy2.py:43]: It should be a TypeVar,
Mypy does not recognize pa.Array, we need a more compact typing system for LazyList, I will do it later.

**Pull request merged**